lace/emulator/pd_archive.py

```diff
@@ -0,0 +1,920 @@
+import numpy as np
+import copy
+import sys
+import os
+import json
+from itertools import product
+import matplotlib.pyplot as plt
+from lace_manager.setup_simulations import read_gadget
+from lace.setup_simulations import read_genic
+from lace.cosmo import camb_cosmo
+from lace.cosmo import fit_linP
+
+
+class archivePD(object):
+    """Book-keeping of flux P1D & P3D measurements from a suite of simulations."""
+
+    def __init__(
+        self,
+        post_processing="768",
+        max_archive_size=None,
+        undersample_z=1,
+        verbose=False,
+        no_skewers=False,
+        pick_sim=None,
+        drop_sim=None,
+        z_max=5.0,
+        nsamples=None,
+        undersample_cube=1,
+        kp_Mpc=None,
+        params_500=True,
+    ):
+        """Load archive from base sim directory and (optional) label
+        identifying skewer configuration (number, width).
+        If kp_Mpc is specified, recompute linP params in archive"""
+
+        # SHOULD UPDATE DOCSTRING WITH ALL THESE ARGUMENTS
+
+        # option_list = [
+        #     "h",
+        #     "nu",
+        #     "central",
+        #     "diffseed",
+        #     "curved",
+        #     "diffigm",
+        #     "running",
+        # ]
+        # for ii in range(31):
+        #     option_list.append(ii)
+        # option_list = np.array(option_list)
+
+        # # check if simulations pick_sim and drop_sim available
+        # if pick_sim is not None:
+        #     if np.any(option_list == pick_sim) == False:
+        #         print(
+        #             "The simulation "
+        #             + str(pick_sim)
+        #             + " is not available. Simulations available:"
+        #         )
+        #         print(option_list)
+        #         raise ValueError("Simulation pick_sim not available")
+        # if drop_sim is not None:
+        #     if np.any(option_list == drop_sim) == False:
+        #         print(
+        #             "The simulation "
+        #             + str(drop_sim)
+        #             + " is not available. Simulations available:"
+        #         )
+        #         print(option_list)
+        #         raise ValueError("Simulation drop_sim not available")
+
+        assert "LACE_REPO" in os.environ, "export LACE_REPO"
+        repo = os.environ["LACE_REPO"] + "/"
+
+        self.post_processing = post_processing
+        if post_processing == "500":
+            self.basedir = "/lace/emulator/sim_suites/Australia20/"
+            self.skewers_label = "Ns500_wM0.05"
+            self.skewers_label_p = "Ns500_wM0.05"
+            self.p1d_label = "p1d"
+            self.p1d_label_p = self.p1d_label
+            self.n_phases = 2
+            self.n_axes = 1
+        elif post_processing == "768":
+            self.basedir = "/lace/emulator/sim_suites/post_768/"
+            self.skewers_label = "Ns768_wM0.05"
+            self.p1d_label = "p1d_stau"
+            if params_500:
```

Comment:
Why do we have this option here? It is a bit strange to have `_500` inside of `_768`.